### PR TITLE
For every variadic function, add a va_list function to the public api

### PR DESCRIPTION
**Pull request by madebr**
By providing these `va_list` alternatives, `SDL_dynapi.c` can be simplified.
It should also help https://github.com/libsdl-org/SDL/issues/9580.

What changed:
SDL_dynapi_proc.h now uses 3 macros:
- `SDL_DYNAPI_PROC`: behavior does not change
- `SDL_DYNAPI_VPROC`: variadic function returning something
- `SDL_DYNAPI_VOID_VPROC`: variadic function returning void

The `*_VPROC` macros accept 2 additional arguments: `forwardfn` and `last_arg`. This is the `va_list ap` function to which it should forward its arguments. The `last_arg` is used as 2nd argument to `va_start`.

I simply appended a `V` to the `va_list` alternatives.

Homework for slouken and icculus:
- come up with better names for the macros and functions
- review whether my changes to the headers are good, and orderered okay (perhaps you want to move all `V` declarations to the bottom)

### CODE REVIEW

**Comment by slouken** [comment]
My initial reaction is that it adds a bunch of entry points to the API and doesn't actually reduce or simplify much code, it just moves the complexity around. In addition, SDL_SetError() is now slightly slower, which is probably okay, but not a great selling point.

@icculus might be super happy about it though, so I'll defer to him. :)

File: src/dynapi/SDL_dynapi.c [hunk 1]
```diff
@@ -213,127 +103,103 @@ static SDL_DYNAPI_jump_table jump_table = {
...
+    static rc SDLCALL fn##_DEFAULT params                                     \
+    {                                                                         \
+        va_list ap;                                                           \
+                                                                              \
+        SDL_InitDynamicAPI();                                                 \
+        va_start(ap, last_arg);                                               \
+        jump_table.forwardfn args;                                            \
+        va_end(ap);                                                           \
+    }
 #define SDL_DYNAPI_PROC_NO_VARARGS 1
```
Comment (madebr) [inline src/dynapi/SDL_dynapi.c:128]: These can go

**Comment by madebr** [comment]
>  In addition, SDL_SetError() is now slightly slower, which is probably okay, but not a great selling point.

I see the contrary on my system (which I don't understand).
Compiling the following source:
```c
#include <SDL3/SDL.h>
#include <SDL3/SDL_main.h>
int main(int argc, char *argv[]) {
    Uint32 i;
    for (i = 0; i < 10000000; i++) {
        SDL_SetError("Time me %" SDL_PRIu32, i);
    }
    return 0;
}
```
and comparing times of SDL3 of current main and this branch, I see faster times on this branch.
I used "hyperfine" to do the benchmark.
I see 1.3s on current main, and 0.9s on this branch.

**Comment by icculus** [comment]
I think pass on this for now.

**Comment by ccawley2011** [comment]
For what it's worth, the MorphOS version of SDL2 provides similar functions to the ones in this PR, which suggests it's neccessary on some platforms.

https://github.com/BeWorld2018/SDL/blob/SDL-2.30.7-release/src/SDL_error.c#L63

**Comment by slouken** [comment]
I added SDL_SetErrorV() in a56778676. All the logging functions can be implemented on top of  SDL_LogMessageV(), so I don't think they're necessary.

**Pull request closed without merging**